lib/openai/assistant_files.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module OpenAI
+  class AssistantFiles
+    class InvalidFileFormat < OpenAI::Error; end
+
+    VALID_EXTENSIONS = %w[
+      .csv .docx .html .java .json .md .pdf .php .pptx .py .rb .tex .txt .css .jpeg .jpg .js .gif
+      .png .tar .ts .xlsx .xml .zip
+    ].freeze
+
+    def initialize(client:, assistant_id:)
+      @client = client.beta(assistants: "v1")
+      @assistant_id = assistant_id
+    end
+
+    def list
+      @client.get(path: "/assistants/#{@assistant_id}/files")
+    end
+
+    def upload(parameters: {})
+      validate(file: parameters[:file])
+
+      @client.multipart_post(
+        path: "/files",
+        parameters: parameters.merge(
+          file: File.open(parameters[:file]),
```

Comment:
I think we should require `file` be an IO object instead of an actual file - thoughts?

Reply:
I think this is a good idea, can you say more about this @allcentury ? What's the reasoning?

Reply:
As an example, I am generating a "File" from a database query.  With this current approach, I have to actually write to a file, then pass it here only to be read again.  That said, in ruby [File#open](https://ruby-doc.org/core-2.5.1/File.html#method-c-open) and other `File` commands returns an `IO` object which is what `Net::HTTP` and other adapters want when sending over HTTP.  

My thought here is that if the caller of this should be able to:

```ruby
@client.upload(file:  File.read("my_file"))

# OR

io = IO.new
output = SomeDatabaseQuery.extract
io.puts output
io.rewind

@client.upload(file: io)
```

In this case, it works for all use cases and users don't need to write temporary files to disk if they won't want to.

Reply:
@allcentury thanks for the suggestion - this has now been added in #420 💨